File: src/day4/part1.py
import re

file = open("input", 'r')
lines = file.readlines()

passports = [passport.replace("\n", "") for passport in ''.join(lines).split("\n\n")]

validentries = ['byr:', 'iyr:', 'eyr:', 'hgt:', 'hcl:', 'ecl:', 'pid:'] #cid:

def passportvalid(passport):
    entries=re.findall('...:', passport)
    return all([x in entries for x in validentries])


print([passportvalid(passport) for passport in passports].count(True))

File: src/day4/haskell/part1.hs
import Data.List.Split
import Data.Text (isInfixOf, pack)
main = do
  input <- readFile "input"
  let l =  map (map (\x -> if x /= '\n' then x else ' ')) $ splitOn "\n\n" input
  let valids = filter (isValid) l
  putStrLn $ show $ length $ filter (isValid) l

validentries = map pack ["byr", "iyr", "eyr", "hgt", "hcl", "ecl", "pid"]
isValid s = all (==True) [x `isInfixOf` (pack s) | x <- validentries]
